core/src/main/java/com/linecorp/armeria/server/ServerMetrics.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2024 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server;
+
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.LongAdder;
+
+/**
+ * A class that holds metrics related server.
+ */
+public final class ServerMetrics {
+
+    private final LongAdder pendingRequests = new LongAdder();
+    private final LongAdder activeRequests = new LongAdder();
+
+    /**
+     * AtomicInteger is used to read the number of active connections frequently.
+     */
+    private final AtomicInteger activeConnections = new AtomicInteger();
+
+    ServerMetrics() {
+    }
+
+    /**
+     * Returns the number of pending requests.
+     */
+    public long pendingRequests() {
+        return pendingRequests.longValue();
+    }
+
+    /**
+     * Returns the number of active requests.
+     */
+    public long activeRequests() {
```

Comment:
What are your thoughts on breaking down `activeRequests` into `activeHttp1Requests`, `activeHttp1WebSocketRequests`, and `activeHttp2Requests`, and then summing them to form `activeRequests`?

Reply:
i think that is a great idea! 
so i added `activeHttp1Requests`, `activeHttp1WebSocketRequests`, and `activeHttp2Request` for `activeRequests`! 
also `pendingHttp1Requests` and `pendingHttp2Requests` for `pendingRequests`